spring-rabbit/src/test/java/org/springframework/amqp/rabbit/log4j2/test/AmqpAppenderTests.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 by the original author(s).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.log4j2.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.FanoutExchange;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageDeliveryMode;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.log4j2.AmqpAppender;
+import org.springframework.amqp.rabbit.test.BrokerRunning;
+import org.springframework.amqp.utils.test.TestUtils;
+
+/**
+ * @author Gary Russell
+ *
+ * @since 1.6
+ */
+public class AmqpAppenderTests {
+
+	@Rule
+	public BrokerRunning brokerRunning = BrokerRunning.isRunning();
+
+	@Test
+	public void test() {
+		CachingConnectionFactory ccf = new CachingConnectionFactory("localhost");
+		RabbitTemplate template = new RabbitTemplate(ccf);
+		RabbitAdmin admin = new RabbitAdmin(ccf);
+		FanoutExchange fanout = new FanoutExchange("log4j2Test");
+		admin.declareExchange(fanout);
+		Queue queue = new Queue("log4jTest");
+		admin.declareQueue(queue);
+		admin.declareBinding(BindingBuilder.bind(queue).to(fanout));
+		Log logger = LogFactory.getLog("foo");
+		logger.info("foo");
+		template.setReceiveTimeout(10000);
+		Message received = template.receive(queue.getName());
+		assertNotNull(received);
+		assertEquals("testAppId.foo.INFO", received.getMessageProperties().getReceivedRoutingKey());
+	}
+
+	@Test
+	public void testProperties() {
+		Log logger = LogFactory.getLog("foo");
+		AmqpAppender appender = (AmqpAppender) TestUtils
+				.getPropertyValue(logger, "logger.context.configuration.appenders", Map.class).get("rabbitmq");
```

Comment:
This works the same way, but more cleaner and without the extra `jcl` dependency:
```
Logger logger = LogManager.getLogger("foo");
AmqpAppender appender = (AmqpAppender) TestUtils
		.getPropertyValue(logger, "context.configuration.appenders", Map.class).get("rabbitmq");
```